tf2/src/buffer_core.cpp

```diff
@@ -1209,13 +1209,18 @@ void BufferCore::cancelTransformableRequest(TransformableRequestHandle handle)
   std::unique_lock<std::mutex> tr_lock(transformable_requests_mutex_);
   std::unique_lock<std::mutex> tc_lock(transformable_callbacks_mutex_);
 
-  V_TransformableRequest::iterator remove_it = std::remove_if(transformable_requests_.begin(), transformable_requests_.end(),
-                                                              [handle](TransformableRequest req) { return handle == req.request_handle; });
-  for (V_TransformableRequest::iterator it = remove_it; it != transformable_requests_.end(); ++it) {
+  V_TransformableRequest::iterator it = std::find_if(
+    transformable_requests_.begin(), transformable_requests_.end(),
+      [handle](TransformableRequest req) { return handle == req.request_handle; });
+  
+  if (it != transformable_requests_.end()) {
     transformable_callbacks_.erase(it->cb_handle);
+    if (transformable_requests_.size() > 1) {
+      transformable_requests_[it -
+        transformable_requests_.begin()] = transformable_requests_.back();
```

Comment:
Can you explain what's happening here?

Imagine the `transformable_requests_.size() == 5`, and the `it` points to the first element, i.e. `it == transformable_requests_.begin()`. Then, once this section is executed, you'd have `transformable_requests_[0] = transformable_requests_.back()`? 

That logic doesn't checkout for me.

Reply:
The solution that I used here is referenced from the 'BufferCore::testTransformableRequests()' function:
https://github.com/ros2/geometry2/blob/16562cee8694c11f1f82b2bdbde2814fca1c7954/tf2/src/buffer_core.cpp#L1305-L1310
The idea is the same: replace the last value to the location on where the value to be deleted, than erase the last element that has been preserved in front.

Reply:
I think I follow what's happening here for the most part, maybe I was wrong in the assumption that there could be multiple requests with the same handle.

Reply:
Certainly, considering the non-unique case increases robustness. But I want to say that the 'addTransformableRequest' function guarantees uniqueness of the 'request_handle' in 'transformable_requests_' vector, unless the sum of the TransformableRequest more than 0xffffffffffffffffULL. If that extreme conditions occur, one handle ID refer to different callbacks, that may cause troubles. So I think non-unique handles cases will not be allowed.